### DISCUSSION TITLE
Submit Form on insert mode doesn't work for me...

### DISCUSSION THREAD

**Discussion by igorgue** [CopilotC-Nvim/CopilotChat.nvim#338] [Q&A]
Hi,

I wonder why `<C-m>` (ctrl+m) doesn't work for me, I wonder how or why this happens would it be my terminal?

```
submit_prompt = {
      normal = '<CR>',
      insert = '<C-m>'
    },
```

**Comment by deathbeam** [comment]
> Hi,
> 
> I wonder why `<C-m>` (ctrl+m) doesn't work for me, I wonder how or why this happens would it be my terminal?
> 
> ```
> submit_prompt = {
>       normal = '<CR>',
>       insert = '<C-m>'
>     },
> ```

most terminals interpret <C-m> as CR, you can verify it by using `showkey -a` and pressing both, they will end up being same keycode so yea, terminal issue for sure

**Comment by igorgue** [comment]
Hmm, yeah this is odd, I use `kitty`, and when I input `CR` on `showkey -a` I get a `^M` which should be the end of line right? But in the plugin it doesn't do anything, I switch to use `<C-CR>` as I'm more used to it in insert mode, but I still find it odd it doesn't work for me, gives me flashbacks of other issues I had before.

**Comment by deathbeam** [comment]
> Hmm, yeah this is odd, I use `kitty`, and when I input `CR` on `showkey -a` I get a `^M` which should be the end of line right? But in the plugin it doesn't do anything, I switch to use `<C-CR>` as I'm more used to it in insert mode, but I still find it odd it doesn't work for me, gives me flashbacks of other issues I had before.

^M should be C-M, im getting same thing. Plugin is just using vim keymaps, maybe you could check if you dont have something already bound to <CR> in insert mode. But yea moving this to discussion as its not rly plugin related
